### PR TITLE
Fix `train_evo2` constant steps description

### PR DESCRIPTION
**Pull request by jannis-baum**
### Description

Fixes the `--help` description of `--constant-steps` in `train_evo2` which says the constant steps happen before cosine annealing but they happen afterwards. You can see the implementation [here](https://github.com/NVIDIA-NeMo/NeMo/blob/main/nemo/lightning/pytorch/optim/lr_scheduler.py) and below is an example of what happens when using the default 80k constant steps:

image

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [x] Documentation update
- [ ] Other (please describe):

### CI Pipeline Configuration

Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

- If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
- If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
train_evo2 --help | grep --after-context=3 "^  --constant-steps"
```

### Pre-submit Checklist

- [x] I have tested these changes locally
- [x] I have updated the documentation accordingly
- [x] I have added/updated tests as needed
- [x] All existing tests pass successfully

## Summary by CodeRabbit

* **Documentation**
  * Clarified CLI help text for the --constant-steps option to say “constant at minimum after annealing” for improved accuracy.
  * Explicitly notes this is a wording change only — no defaults, functionality, or runtime behavior were modified.

### CODE REVIEW

**Comment by jannis-baum** [comment]
Hello! Just bumping this since it seems like it was missed. @pstjohn @jstjohn tagging you as you were the ones who approved #907 which introduced the incorrect description.

**Comment by jannis-baum** [comment]
Is there anything missing from my side for this to be merged? If not, what are we waiting for? It's a one-line fix to a help text that says the opposite of what it should in the current version.

**Comment by jstjohn** [comment]
Just put it in the queue.

**Comment by trvachov** [comment]
Oh geez -- we let this one slide through the cracks. Apologies @jannis-baum  and thanks for the contribution.

**Comment by trvachov** [comment]
@jannis-baum sorry again -- thanksgiving holidays have held this PR up. I'll merge today 🥳

**Comment by trvachov** [comment]
@pstjohn not sure what's going on with CI right now -- force merging.
